### ISSUE TITLE
Navigation section titles

### ISSUE THREAD

**Issue by mattnolting** [patternfly/patternfly-react#6770]
Related to: https://github.com/patternfly/patternfly/issues/4602

Need to make visible section titles optional. If section title not visible, section title text should be applied to to `aria-label`.

**Comment by wise-king-sullyman** [comment]
There are two primary approaches to this that come to my mind:

1. an `isTitleVisible` prop that when true sets the passed `title` as the title, and when false sets the `title` as an aria-label instead (this is the more direct/literal interpretation of the ticket as I read it).

1. making the current `title` prop optional and only rendering the title `h2` when a `title` is passed; then just letting consumers pass an `aria-label` manually as they would now. I lean towards this approach because I believe this behavior is simpler/clearer. The primary downside to this approach is that it _does_ create the possibility that a nav group has no `title` or `aria-label` if the consumer doesn't apply either prop.

Does anyone have any particular feelings about either of those approaches? If not I'll be implementing the second approach.

**Comment by mattnolting** [comment]
@wise-king-sullyman could you add a fallback `aria-title` value? As in, index the `nav-group`s inside `nav` and automatically generate `Section title {index}`? As an aside, I wouldn't present this in an example..

**Comment by wise-king-sullyman** [comment]
@mattnolting because of the composable nature of `nav` (with the `nav-group`s being children that the consumer passes to `nav`) I think the only clean way to do that sort of thing would have to be on the consumer end, which I think would kind of defeat the point because then it isn't "automatic" anyways.

From a somewhat quick look, I think something could probably be done using the `nav` context, having `nav-group`s register their existence with `nav`, and getting assigned an index value that way, but that would be adding a good bit of complication for a fallback behavior. 

I'm not saying I'm averse to trying to figure out how to do so, but I don't think the benefits would outweigh the cost.

**Comment by thatblindgeye** [comment]
Personally, my initial thoughts were in agreement with the first option Austin suggested: creating an `isTitleVisible` prop and if set to false, passing the value of the required `title` prop to the aria-label. My main reason being that, without a visible section/group title there may still be visual indication that there are groups of items (either that horizontal divider or the expandable buttons), but using a screen reader some context _might_ be lost (especially when navigating via screen reader shortcuts).

That said, this might be something for a larger discussion first (forcing the decision on an `aria-label`'s value vs keeping it as the consumer's decision). If that would be the case, I would say Austin's second suggestion (making the `title` prop optional and allowing the consumer the option to pass in an `aria-label`) in addition to some example instructions regarding adding an aria-label would probably work better for now.

**Comment by mcoker** [comment]
> The primary downside to this approach is that it does create the possibility that a nav group has no title or aria-label if the consumer doesn't apply either prop.

Could this produce a warning/error if neither are set?
